src/common/expression/LogicalExpression.h

```diff
@@ -79,6 +79,16 @@ class LogicalExpression final : public Expression {
 
   bool isLogicalExpr() const override { return true; }
 
+  void reverseLogicalKind() {
+    if (kind_ == Kind::kLogicalAnd) {
+      kind_ = Kind::kLogicalOr;
+    } else if (kind_ == Kind::kLogicalOr) {
+      kind_ = Kind::kLogicalAnd;
+    } else {
+      LOG(FATAL) << "Should not reverse logical expression except and/or kind.";
+    }
+  }
+
```

Comment:
We have `static LogicalExpression* reverseLogicalExpr(LogicalExpression* expr);` defined in `ExpressionUtils.h`. Maybe you could utilize it.

Reply:
It's not suitable to use reverseLogicalExpr in ExpressionUtils.h here. Because here we just need to reverse the logical kind while maintaining the same operands. However, that reverseLogicalExpr function will makeNot every operand.